native/src/main/java/io/ballerina/wso2/apim/catalog/ServiceCatalog.java

```diff
@@ -0,0 +1,263 @@
+        updateMd5(artifactValues);
+    }
+
+    private static void updateMd5(BMap<BString, Object> artifactValues) {
+        String name = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(NAME)));
+        String version = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(VERSION)));
+        String definitionType = StringUtils.getStringValue(artifactValues.get(StringUtils.fromString(DEFINITION_TYPE)));
+        String openapiDef = StringUtils.getStringValue(artifactValues.get(StringUtils
+                .fromString(DEFINITION_FILE_CONTENT)));
+
+        String string = new StringBuilder(name).append(version).append(definitionType).append(openapiDef).toString();
+        artifactValues.put(StringUtils.fromString(MD5), StringUtils.fromString(createMd5Hash(string)));
+    }
+
+    private static void updateServiceNameAndUrl(BMap<BString, Object> artifactValues,
+                                                HttpServiceConfig httpServiceConfig) {
+        updateServiceName(artifactValues, httpServiceConfig);
+        updateServiceUrl(artifactValues, httpServiceConfig);
+    }
+
+    private static void updateListenerConfigurations(BMap<BString, Object> artifactValues, Object listenerDetails) {
+        artifactValues.put(StringUtils.fromString(MUTUAL_SSL_ENABLED), getMutualSSLDetails(listenerDetails));
+    }
+
+    private static boolean getMutualSSLDetails(Object listenerDetails) {
+        ArrayList<BObject> listenerArray = (ArrayList<BObject>) listenerDetails;
+        if (!listenerArray.isEmpty()) {
+            BObject listener = listenerArray.get(0);
+            if (listener != null) {
+                Object configObject = listener.getNativeData(CONFIG);
+                if (configObject instanceof HashMap config
+                        && config.containsKey(StringUtils.fromString(SECURE_SOCKET))) {
+                    Object secureSocketObject = config.get(StringUtils.fromString(SECURE_SOCKET));
+                    return secureSocketObject instanceof HashMap secureSocket
+                            && secureSocket.containsKey(StringUtils.fromString(MUTUAL_SSL));
+                }
+            }
+        }
+        return false;
+    }
+
+    private static void updateServiceName(BMap<BString, Object> artifactValues, HttpServiceConfig httpServiceConfig) {
+        artifactValues.put(StringUtils.fromString(NAME),
+                StringUtils.fromString(httpServiceConfig.basePath));
+    }
+
+    private static void updateServiceUrl(BMap<BString, Object> artifactValues, HttpServiceConfig httpServiceConfig) {
+        String basePath = httpServiceConfig.basePath.equals(LOCALHOST) ? "" : httpServiceConfig.basePath;
+        artifactValues.put(StringUtils.fromString(SERVICE_URL),
+                StringUtils.fromString(new StringBuilder().append(httpServiceConfig.host).
+                        append(COLON).append(httpServiceConfig.port).append(basePath).toString()));
```

Comment:
Assuming intention here is to concat the string just directly use `+`. In older JVM (I belive 6+) javac will convert it to use `StringBuilder` anyway and newer ones will use the more efficient `makeConcatWithConstant`